pypom/page.py

```diff
@@ -2,19 +2,63 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
+try:
+    from urlparse import urljoin
+except ImportError:
+    from urllib.parse import urljoin
+
 from .view import WebView
 
 
 class Page(WebView):
 
-    _url = '{base_url}'
+    URL_TEMPLATE = ''
     """String representing a URL that will return this page.
 
     This string is formatted and can contain names of keyword arguments passed
-    during construcion of the page object. The ``{base_url}`` keyword argument
-    is always passed to the format operation.
+    during construction of the page object. The template should either assume
+    that the value of ``base_url`` will be prepended to the result of the
+    format, or should yield an absolute URL.
     """
 
+    @property
+    def target_url(self):
+        """
+        Returns the URL to the current page,
+        formatted from :py:data:`URL_TEMPLATE`.
+        Unless URL_TEMPLATE results in an absolute URL, ``self.base_url`` is
+        prepended to the resulting URL.
+
+        :returns:
+            String representing a URL that will return this page.
+        """
+        return urljoin(self.base_url,
+                       self.URL_TEMPLATE.format(**self.url_kwargs))
```

Comment:
What I like about `urljoin` is that it has URL-smarts built in. It treats the first URL as the default for all unspecified parts of the second URL, so if the second URL is already absolute it just ignores the first URL. If the first URL is empty then it will ignore that and return the second URL unchanged. I believe it does exactly what we want, but it is causing us this coverage issue because of the two imports.

I'm going to change the code to just check for the presence of `//` in the result of the format and if that is present then assume we already have an absolute URL.